[2025-06-02T05:53:07.730+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: movie_recommendation_pipeline.dvc_add_and_push_all manual__2025-06-02T05:37:26.854773+00:00 [queued]>
[2025-06-02T05:53:07.737+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: movie_recommendation_pipeline.dvc_add_and_push_all manual__2025-06-02T05:37:26.854773+00:00 [queued]>
[2025-06-02T05:53:07.737+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 3
[2025-06-02T05:53:07.748+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): dvc_add_and_push_all> on 2025-06-02 05:37:26.854773+00:00
[2025-06-02T05:53:07.754+0000] {standard_task_runner.py:60} INFO - Started process 755 to run task
[2025-06-02T05:53:07.756+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'movie_recommendation_pipeline', 'dvc_add_and_push_all', 'manual__2025-06-02T05:37:26.854773+00:00', '--job-id', '1509', '--raw', '--subdir', 'DAGS_FOLDER/mlops_movie_dag.py', '--cfg-path', '/tmp/tmpudex_2ax']
[2025-06-02T05:53:07.757+0000] {standard_task_runner.py:88} INFO - Job 1509: Subtask dvc_add_and_push_all
[2025-06-02T05:53:07.791+0000] {task_command.py:423} INFO - Running <TaskInstance: movie_recommendation_pipeline.dvc_add_and_push_all manual__2025-06-02T05:37:26.854773+00:00 [running]> on host 0e6e1ff1f8df
[2025-06-02T05:53:07.851+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='movie_recommendation_pipeline' AIRFLOW_CTX_TASK_ID='dvc_add_and_push_all' AIRFLOW_CTX_EXECUTION_DATE='2025-06-02T05:37:26.854773+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-06-02T05:37:26.854773+00:00'
[2025-06-02T05:53:07.854+0000] {mlops_movie_dag.py:12} INFO - üü¶ Running command: git config --global --add safe.directory /opt/***
[2025-06-02T05:53:07.863+0000] {mlops_movie_dag.py:19} INFO - üü© [stdout]:

[2025-06-02T05:53:07.863+0000] {mlops_movie_dag.py:12} INFO - üü¶ Running command: git config --global user.email ***@mlops.local
[2025-06-02T05:53:07.882+0000] {mlops_movie_dag.py:19} INFO - üü© [stdout]:

[2025-06-02T05:53:07.882+0000] {mlops_movie_dag.py:12} INFO - üü¶ Running command: git config --global user.name Airflow
[2025-06-02T05:53:07.900+0000] {mlops_movie_dag.py:19} INFO - üü© [stdout]:

[2025-06-02T05:53:07.900+0000] {mlops_movie_dag.py:12} INFO - üü¶ Running command: dvc add /opt/***/models/hybrid_model.pkl
[2025-06-02T05:53:22.340+0000] {mlops_movie_dag.py:19} INFO - üü© [stdout]:

To track the changes with git, run:

	git add models/hybrid_model.pkl.dvc

To enable auto staging, run:

	dvc config core.autostage true

[2025-06-02T05:53:22.341+0000] {mlops_movie_dag.py:12} INFO - üü¶ Running command: dvc add /opt/***/data/processed/hybrid_matrix.csv
[2025-06-02T05:53:39.557+0000] {mlops_movie_dag.py:19} INFO - üü© [stdout]:

To track the changes with git, run:

	git add data/processed/hybrid_matrix.csv.dvc

To enable auto staging, run:

	dvc config core.autostage true

[2025-06-02T05:53:39.558+0000] {mlops_movie_dag.py:12} INFO - üü¶ Running command: dvc add /opt/***/data/processed/hybrid_matrix_features.txt
[2025-06-02T05:53:50.243+0000] {mlops_movie_dag.py:19} INFO - üü© [stdout]:

To track the changes with git, run:

	git add data/processed/hybrid_matrix_features.txt.dvc

To enable auto staging, run:

	dvc config core.autostage true

[2025-06-02T05:53:50.243+0000] {mlops_movie_dag.py:12} INFO - üü¶ Running command: dvc add /opt/***/models/user_model.pkl
[2025-06-02T05:53:53.165+0000] {mlops_movie_dag.py:19} INFO - üü© [stdout]:

[2025-06-02T05:53:53.166+0000] {mlops_movie_dag.py:21} WARNING - üü® [stderr]:
ERROR: Unable to acquire lock. Most likely another DVC process is running or was terminated abruptly. Check the page <https://dvc.org/doc/user-guide/troubleshooting#lock-issue> for other possible reasons and to learn how to resolve this.

[2025-06-02T05:53:53.166+0000] {mlops_movie_dag.py:29} ERROR - ‚ùå Subprozess-Fehler (exit code 1): dvc add /opt/***/models/user_model.pkl
[2025-06-02T05:53:53.167+0000] {mlops_movie_dag.py:32} ERROR - ‚ùå Subprozess-Ausnahme: Command '['dvc', 'add', '/opt/***/models/user_model.pkl']' returned non-zero exit status 1.
[2025-06-02T05:53:53.167+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/mlops_movie_dag.py", line 90, in dvc_add_and_push_all
    run_and_log(["dvc", "add", art])
  File "/opt/airflow/dags/mlops_movie_dag.py", line 30, in run_and_log
    raise subprocess.CalledProcessError(result.returncode, command)
subprocess.CalledProcessError: Command '['dvc', 'add', '/opt/airflow/models/user_model.pkl']' returned non-zero exit status 1.
[2025-06-02T05:53:53.181+0000] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=movie_recommendation_pipeline, task_id=dvc_add_and_push_all, execution_date=20250602T053726, start_date=20250602T055307, end_date=20250602T055353
[2025-06-02T05:53:53.197+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 1509 for task dvc_add_and_push_all (Command '['dvc', 'add', '/opt/airflow/models/user_model.pkl']' returned non-zero exit status 1.; 755)
[2025-06-02T05:53:53.206+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-06-02T05:53:53.226+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
